### PR TITLE
Added persistent score storage and gestion

### LINKED ISSUE #15: Make high scores persistent
* make all high scores persist after the player leaves the game

### LINKED ISSUE #16: High score logic
* keep tally of the high scores
* allow player to put their name on their new high score?

### PR DESCRIPTION
**Pull request by f-michaut**
- Added .vsconfig to gitignore
- Created ScoreManager class
- Added functions to add new scores, list the current ones, and clear the scores
- At startup, load the score list from a file
- All the functions that change the score list write the new content to the disk

### CODE REVIEW

File: Assets/Code/Scripts/ScoreManager.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
+
+public class ScoreManager : MonoBehaviour
+{
+    [Serializable]
+    public class ScoreData
```
Comment (ClaudiaLapalme) [inline Assets/Code/Scripts/ScoreManager.cs:11]: I didn't realize that you could serialize a class. This is very cool and useful for testing

File: Assets/Code/Scripts/ScoreManager.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            return new List<ScoreData>();
+        }
+    }
+
+    // Return true if the given score will enter the highscore list
+    public static bool IsHighScore(int newscore)
+    {
+        if (scores.Count < maxHighscores)
+            return true;
+        foreach (ScoreData score in scores) {
```
Comment (ClaudiaLapalme) [inline Assets/Code/Scripts/ScoreManager.cs]: Wouldn't it be more efficient to just check the last value (or first, depending on if it's ascending or descending). I noticed that you already sort when a new high score is entered anyways.

File: Assets/Code/Scripts/ScoreManager.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        return false;
+    }
+
+    // Add the new highscore to list and sort it
+    public static void AddScore(string name, int score)
+    {
+        if (!IsHighScore(score))
+            return;
+        scores.Add(new ScoreData { date = System.DateTime.Now, name = name, score = score });
+        scores.Sort((ScoreData a, ScoreData b) => { return a.score - b.score; });
```
Comment (ClaudiaLapalme) [inline Assets/Code/Scripts/ScoreManager.cs:60]: I'm a little confused by the return on this line. What does it do exactly?

**Review by ClaudiaLapalme — CHANGES_REQUESTED** [review]

File: Assets/Code/Scripts/ScoreManager.cs (hunk 2 above)
Comment (f-michaut) [reply Assets/Code/Scripts/ScoreManager.cs]: Oh you are right, i didn't think of taking adventage of the sorting 
I think i just need to check if the value is comprised between the first and last values
I'll change that

File: Assets/Code/Scripts/ScoreManager.cs (hunk 3 above)
Comment (f-michaut) [reply Assets/Code/Scripts/ScoreManager.cs:60]: It's the return of the lambda function the Sort function will use to compare values in the list. It return an integer indicating which value is the biggest. So it return an integer less than, equal to or bigger than 0 following the same convention as in this page : https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.comparer-1.compare?view=net-5.0

File: Assets/Code/Scripts/ScoreManager.cs (hunk 1 above)
Comment (f-michaut) [reply Assets/Code/Scripts/ScoreManager.cs:11]: You can serialize a class, and most of c#'s containers are also serializable, meaning you can serialize a class containing Lists or Disctionary of other classes. Altho, i think there is some restrictions, i don't remember them exactly tho

**Review by ClaudiaLapalme — APPROVED** [review]
great work!

**Review by tedgag — APPROVED** [review]
Looks good! Approved

**Pull request merged**